### ISSUE TITLE
[$500] [Medium] TRIPPREVIEW is not showing the preview/info

### ISSUE THREAD

**Issue by kevinksullivan** [Expensify/App#45348]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**  https://expensify.slack.com/archives/C05S5EV2JTX/p1720621730730619

## Action Performed:
1. Book a trip in spotnana
2. Navigate to the workspace chat

## Expected Result:
Preview should load/show info

## Actual Result:
Preview is not reliably loading/showing information

https://files.slack.com/files-pri/T047TPA624F-F07BMUQ00P8/screenshot_2024-07-10_at_7.58.02___pm.png

## Screenshots/Videos

![image](https://github.com/user-attachments/assets/6a560444-0983-4271-b134-8c2606258a98)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014f530c69599f191f</li>
        <li>Upwork Job ID: 1812949990970484758</li>
        <li>Last Price Increase: 2024-09-30</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @johncschuster</details>

**Comment by johncschuster** [comment]
Given the urgency, I've opted to assign this to the engineers curating the `[#travel-gbta-july-22-2024](https://expensify.enterprise.slack.com/archives/C05S5EV2JTX)` project. Please reassign as needed!

**Comment by johncschuster** [comment]
[Bumping Slack for proposals](https://expensify.slack.com/archives/C01GTK53T8Q/p1721337034940629)

**Comment by twisterdotcom** [comment]
Hmm, I thought that would double it.

**Comment by b4s36t4** [comment]
Reproduction steps are unclear, is there any video to demonstrate the issue clearly?!

**Comment by johncschuster** [comment]
@b4s36t4 I see you requested in Slack to be added to the beta. I've just done that now.

**Comment by johncschuster** [comment]
Thanks for the heads up, @b4s36t4! I'll keep an eye on that issue as well.

**Comment by kevinksullivan** [comment]
Hi @b4s36t4 can you try again?

**Comment by b4s36t4** [comment]
@kevinksullivan it is still same, while this time I even got error for Accepting the Spotana Terms as well.
<img width="530" alt="Screenshot 2024-07-25 at 5 35 34 PM" src="https://github.com/user-attachments/assets/531b249c-44d3-4242-9373-a772eec6b921">
<img width="535" alt="Screenshot 2024-07-25 at 5 35 41 PM" src="https://github.com/user-attachments/assets/4c073e7d-6cee-4d7f-8030-3c147988f6c7">

Is it fixed in the backend or should I be using a new account for testing?

**Comment by johncschuster** [comment]
@kevinksullivan as far as I can tell, this issue is kinda blocked on https://github.com/Expensify/App/issues/45647 – have I understood that correctly?

**Comment by johncschuster** [comment]
Looks like https://github.com/Expensify/App/issues/45647 has been resolved! @b4s36t4 can you try one more time?

**Comment by kevinksullivan** [comment]
Bump @b4s36t4

**Comment by kevinksullivan** [comment]
also increasing $ amount since this is a CRITICAL item for travel and we want eyes on this asap.

**Comment by chrispader** [comment]
Are there any more detailed steps on how to "Book a trip in spotnana"? cc @kevinksullivan

**Comment by stitesExpensify** [comment]
Hey @chrispader we would need to add your account to the beta. I can do that for you if you want to give me the email address associated :)

**Comment by twisterdotcom** [comment]
Added you to the beta. You should see it in an hour or so.

**Comment by kevinksullivan** [comment]
updating priority per [this convo](https://expensify.slack.com/archives/C05S5EV2JTX/p1722988187474099) to focus on OldDot first.

**Comment by stitesExpensify** [comment]
@cretadn22 this is the test account setup steps


```
If you are creating a new test account after 07/29/2024
Create a new Expensify Account on OldDot

Currently travel is on a beta, so use expensifail, or add the account to the beta manually at expensify.com/_support/beta
Open the console and paste this command NVP.set('travelSettings', {'testAccount': true});

Refresh the browser

You can now use this account on Production or Staging to test/demo via the sandbox

Note:
Once an account is provisioned (you accept the terms and are sent to Spotnana for the first time) that account and policy are both now permanently a production or test account depending on if the NVP was set or not. You cannot switch between them. Additionally if you add more users to the policy, they will need to have the NVP set on their account as well!
```

**Comment by stitesExpensify** [comment]
> 2.1: When going to expensify.com/_support/beta , It redirect to expensify.com/inbox

This is expected, it is an internal tool. Ted [did that step for you](https://github.com/Expensify/App/issues/45348#issuecomment-2275337063) :) 

> The API still return error on NewDot: User is not on Travel Beta

Hm are you sure that you're logged into the right account? You're definitely on the beta with cretadn22@gmail.com
<img width="1723" alt="2024-08-12_10-13-45" src="https://github.com/user-attachments/assets/741dd0df-4431-40d4-a50c-1e6dbf6d93eb">

**Comment by stitesExpensify** [comment]
Hm you're getting `permission denied`. Is it possible that you signed up for travel _before_ adding the NVP in OldDot, and then tried again?

**Comment by stitesExpensify** [comment]
Yes that is correct. I will add your account to the beta now. Please wait ~ 1 hour before continuing with steps 3/4

**Comment by SofiedeVreese** [comment]
Hi @cretadn22 I can see you've been booking in a few demos with our team. 

When you do, can you please ensure you add `testing` to your name so our team knows it's not a real demo? We're back-to-back demo'ing to prospective customers so this is filling up precious time in our calendars we can use for real customers. If you add `testing` we'll know it's a time we'll know to either cancel the call or simply use that time for other calls!

**Comment by stitesExpensify** [comment]
Hi @cretadn22 you should be able to use a stripe test card https://docs.stripe.com/testing (4242 specifically)

**Comment by johncschuster** [comment]
@stitesExpensify bump on the ^previous comment

**Comment by stitesExpensify** [comment]
Whoops so sorry that I missed this. @cretadn22 try booking SFO to JFK on Alaskan airlines. The sandbox can be finicky and that one seems to work the best

**Comment by NJ-2020** [comment]
@stitesExpensify Can you please add my account to beta?
My email is: novellj123@gmail.com

**Comment by ChavdaSachin** [comment]
I am interested in this issue
@stitesExpensify Can you please add my account to beta?
My email is: sachin.chavda.1024@gmail.com

**Comment by stitesExpensify** [comment]
Added you both. Please give it ~1 hour to take effect.

Make sure to run this in the console on oldDot before you click the travel button for the first time or your account will be locked to production (which costs real money) instead of the sandbox.
`NVP.set('travelSettings', {'testAccount': true});`

**Comment by ChavdaSachin** [comment]
I am facing the same issue as @cretadn22, I successfully booked several trips including flight and car trips and did not receive any relevant message either on New dot nor Old dot.

**Comment by NJ-2020** [comment]
@stitesExpensify Could you please accept my trips? Thanks

**Comment by stitesExpensify** [comment]
@ChavdaSachin i will look into the logs and see if we got any events for your trips.

@NJ-2020 what do you mean by accepting your trips? Am I set as your manager in Spotnana?

**Comment by ChavdaSachin** [comment]
Tnx @stitesExpensify. But until you check the log and probably find and deal with the BE/sandbox issue, if you could provide me an example onyx object for a trip then I could probably mock it and figure my way around.

**Comment by NJ-2020** [comment]
@stitesExpensify I mean, after we booked the trips, we see like waiting for confirmation right?
I mean is there any way like you can confirm our trips

**Comment by stitesExpensify** [comment]
> I mean, after we booked the trips, we see like waiting for confirmation right?
> I mean is there any way like you can confirm our trips

The confirmation being referred to is the airline confirmation not something that I can do unfortunately. And on staging the trip will never be confirmed because no real tickets are being issued. 

You could maybe try booking a trip further out like a month or 2? If that doesn't work I'll try to get you some onyx data that does not include PII

**Comment by stitesExpensify** [comment]
@NJ-2020 did that work?

**Comment by ChavdaSachin** [comment]
@stitesExpensify I tried it as well, didn't workout

**Comment by Gonals** [comment]
Do we have screenshots/mockups of how this should look? (Has this ever worked?)

**Comment by johncschuster** [comment]
Not overdue, Melv. Thanks for taking a look, @shubham1206agra!

**Comment by johncschuster** [comment]
@shubham1206agra were you able to find anything?

**Comment by iwiznia** [comment]
Discussions ongoing, not overdue

**Comment by johncschuster** [comment]
Looks like we're waiting on @stitesExpensify in that Slack thread. I've bumped him in NewDot 👍

**Comment by iwiznia** [comment]
Not yet @stitesExpensify [needs to keep looking](https://expensify.slack.com/archives/C05S5EV2JTX/p1727108536185879?thread_ts=1725546707.583219&cid=C05S5EV2JTX)

**Comment by johncschuster** [comment]
Same as above

**Comment by stitesExpensify** [comment]
Hey everyone, I'm removing the help wanted label from this for now. We are currently focusing all efforts on Travel in OldDot, so this has been shelved until we have released that feature fully, or have a lot more hands on deck. Thank you everyone who investigated this, and hopefully we can get back to this soon!

**Comment by stitesExpensify** [comment]
Not overdue

**Comment by johncschuster** [comment]
_Still_ not overdue, Melvin 🤦

**Comment by cristipaval** [comment]
No update yet, I'll hopefully have an update before the end of the day.

**Comment by cristipaval** [comment]
I could reproduce the issue as follows:
1. Have an account with `focus` set for the LHN
2. Have a trip preview in a workspace chat
3. Sign out and then sign in again
4. Go to the workspace chat and verify the trip preview. It doesn't show the details of the trip.

The root cause seems to be that the OpenApp command returns the `tripRoom` report linked to the workspace chat, but it doesn't return the report with the `iouReportID` set on the trip report. Once the user first opens the trip and fetches that report, everything works fine. I'll continue investigating to confirm this theory and fix the issue.

**Comment by cristipaval** [comment]
Oh, forgot to post an update.

I investigated the issue even more and it seems that there are cases when our webhook is called by Spotnana with eventType and operation combinations that are not fully handled in our core. This leads to corner cases when we create the trip room but we don't create the expense report linked to it. Such a case is when the user creates a pre-paid booking but the payment is declined. More info [here](https://expensify.slack.com/archives/C05S5EV2JTX/p1732715888160519)

I'll open today a fix PR for this.

**Comment by cristipaval** [comment]
Not overdue, Melv

**Comment by johncschuster** [comment]
Not overdue. @cristipaval's working on a PR

**Comment by cristipaval** [comment]
The [Auth PR](https://github.com/Expensify/Auth/pull/13394) is ready for review.

**Comment by cristipaval** [comment]
Fix PR hit production. Closing

**Comment by cristipaval** [comment]
@johncschuster could you please check the above comment? 🙏

**Comment by johncschuster** [comment]
@shubham1206agra can you help me understand what the compensation would be for? Would it be for reporting the issue, or did you contribute to the PR that led to the resolution?

If the latter, can you point out where you contributed? That'll make it way easier to review given the length of this issue. Thank you!

**Comment by cristipaval** [comment]
I also participated in that [Slack thread](https://github.com/Expensify/App/issues/45348#issuecomment-2342428608), where we chatted about the steps to reproduce the issue. I think we usually do a partial pay when the C+ reviews proposals before we decide to take the issue internally. We also do this when we accept a proposal, the contributor starts working on it, and then we decide to close the issue or fix it internally. Other than the bug report in Slack, I can't tell concrete contributions from @shubham1206agra to the solution, @johncschuster, so I'm not sure if the #bug-zero pays anything in these situations.
